src/UI/Art/Animation/GameAnimation.java

```diff
@@ -13,4 +28,23 @@
    * Allows the caller to tell this animation to end early.
    */
   public void cancel();
+
+  public default Unit getActor()
```

Comment:
These changes break the interface. `GameAnimation` should be usable for CO abilities, in-game events (e.g. volcanic eruptions), and whatever else. Pushing unit-specific (and worse, action-type-specific) information to the top level ruins the abstraction. Eventually, we'll just have to keep adding things to support more stuff for different animation types.

Reply:
The `MapView` needs to know whether/where to draw the current acting unit.
`MapView` doesn't have a pre-existing concept of volcano eruptions, so it won't need instructions to modify how it draws any particular one. Units, OTOH, are always being drawn.
This is a special case that's broadly applicable and (because of default implementations) won't require animations that don't need it to put code in to support it.

The only fix I can see is making move and combat animations draw the unit themselves, and still report `getActor()` so the `MapView` knows not to draw the actor itself.

Reply:
> The only fix I can see

Hold my Kool-aid.